Frameworks/MQTTnet.NetFramework/MqttServerFactory.cs

```diff
@@ -8,8 +9,9 @@ public class MqttServerFactory
         public MqttServer CreateMqttServer(MqttServerOptions options)
         {
             if (options == null) throw new ArgumentNullException(nameof(options));
-
-            return new MqttServer(options, new MqttServerAdapter());
+            
+            // The cast to IMqttServerAdapter is required... stupidly...
+            return new MqttServer(options, options.UseSSL ? (IMqttServerAdapter) new MqttSslServerAdapter() : new MqttServerAdapter());
```

Comment:
I have some doubts because with this code either encryted or not encrypted will be supported. I have some device which can be encrypted but others are not and I don't have access to the firmware. I can just start 2 servers but then the clients are separated from each other. Maybe a list of adapters is a god solution to fix this.  So the options are having a boolean "ExposeSslEndpoint" and "ExposeUnencryptedEndpoint" and "EncryptedEndpointPort" and "UnencryptedEndpointPort" etc. and depending on this the factory can create a list of adapters with either both or only one. But this only  a draft of an idea :-) What do you think about this idea?